posix/unix.c

```diff
@@ -698,6 +713,11 @@ static ssize_t send(unsigned socket, const void *buf, size_t len, int flags, con
 				err = -ENOTCONN;
 				break;
 			}
+			else if (s->state & US_DISCONNECTED) {
```

Comment:
not sure, but it seems that we will be sending `SIGPIPE` on every `send()`, maybe we should instead send it once and set `s->connect = NULL` here and and clear `US_DISCONNECTED` flag?

Reply:
Linux sends SIGPIPE on every send.